samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -1184,17 +1316,39 @@ int main(void)
 			continue;
 		}
 
-		sync_bitfield = bis_index_bitfield & requested_bis_sync;
-		stream_count = 0;
-		for (int i = 0; i < BT_ISO_MAX_GROUP_ISO_COUNT; i++) {
-			if ((sync_bitfield & BIT(i)) != 0) {
-				stream_count++;
-			}
+		/* Only base BIS indexes from requested subgroups are of interest */
+		base_bis_index_bitfield &= get_subgroup_mask(&bis_index, requested_subgroup_sync);
+
+		/* If requested BIS indexes are 'no_pref' then adjust number of BIS indexes to
+		 * number of streams supported
+		 */
+		if (requested_bis_sync == BT_BAP_BIS_SYNC_NO_PREF) {
+#if defined(CONFIG_TARGET_BROADCAST_CHANNEL)
+			/* Keep the n least significant bits of base_bis_index_bitfield to match the
+			 * target allocation
+			 */
+			base_bis_index_bitfield = keep_least_significant_allocation_match(
+				base_bis_index_bitfield, &bis_index.audio_allocation[0],
+				CONFIG_TARGET_BROADCAST_CHANNEL);
```

Comment:
Since `bis_index` and `CONFIG_TARGET_BROADCAST_CHANNEL` are global, there's not really any purpose of providing them as arguments